src/ccl_halomod.c

```diff
@@ -0,0 +1,300 @@
+#include "ccl_utils.h"
+#include "math.h"
+#include "stdio.h"
+#include "stdlib.h"
+#include "gsl/gsl_integration.h"
+#include "gsl/gsl_interp.h"
+#include "gsl/gsl_spline.h"
+#include "gsl/gsl_sf_expint.h"
+#include "ccl_placeholder.h"
+#include "ccl_background.h"
+#include "ccl_power.h"
+#include "ccl_massfunc.h"
+#include "ccl_error.h"
+#include "class.h"
+#include "ccl_params.h"
+#include "ccl_emu17.h"
+#include "ccl_emu17_params.h"
+
+double delta_c(){
+  // Linear collapse threshold for haloes
+  return 1.686;
+}
+
+// TODO: referred to as odelta in ccl_massfunc, as a free parameter.
+double Delta_v() {
+  // Halo mean density
+  return 200.;
+}
+
+// TODO: Check growth factor is normalised g(a=1)=1
+double k_star(ccl_cosmology *cosmo, double a, int * status){
+  // HMcode parameter: k*  
+  double sigmaVz = ccl_sigmaV(cosmo, 0., status)*ccl_growth_factor(cosmo, a, status);
+  return 0.584/sigmaVz;
+}
+
+// TODO: Check growth factor is normalised g(a=1)=1
+double f_damp(ccl_cosmology *cosmo, double a, int * status){
+  // HMcode parameter: f  
+  double sigma8z = ccl_sigma8(cosmo, status)*ccl_growth_factor(cosmo, a, status);
+  return 0.188*pow(sigma8z, 4.29);
+}
+
+//TODO: Actually code this up
+double collapse_index(ccl_cosmology *cosmo, double a, int * status){
+  // HMcode parameter: spectral index at the collapse scale, neff
+  return -2.;
+}
+
+double alpha(ccl_cosmology *cosmo, double a, int * status){
+  // HMcode parameter: alpha to transition between one- and two-halo regimes
+  double neff = collapse_index(cosmo, a, status);
+  return 2.93*pow(1.77,neff);
+}
+
+double comoving_matter_density(ccl_cosmology *cosmo){
+  // The comoving density of matter. This is a constant. Units are Msun/Mpc^3 (no factors of h)
+  return RHO_CRITICAL*cosmo->params.Omega_m*cosmo->params.h*cosmo->params.h;
+}
+
+// TODO: possible that delta should be passed around for consistency checks
+double r_delta(ccl_cosmology *cosmo, double halomass, double a, int * status){  
+  // Converts halo mass to rdelta.
+  double rho_matter = comoving_matter_density(cosmo);  
+  //rho_m = RHO_CRITICAL*cosmo->params.Omega_m*cosmo->params.h*cosmo->params.h;  
+  return pow(halomass*3.0/(4.0*M_PI*rho_matter*Delta_v()),1.0/3.0);
+}
+
+double r_Lagrangian(ccl_cosmology *cosmo, double halomass, double a, int * status){
+  // Calculates the halo Lagrangian radius as a function of halo mass
+  double rho_matter = comoving_matter_density(cosmo);  
+  return pow(halomass*3.0/(4.0*M_PI*rho_matter),1.0/3.0);
+}
+
+// TODO: Why is 'sinl' used in the below routine as well as 'sin'?
```

Comment:
in case this question is based on my code snippets: you're free to use either, I had just found sinl to be more stable for small arguments.